static/app/views/insights/pages/frontend/frontendPageHeader.tsx

```diff
@@ -0,0 +1,99 @@
+import {Fragment} from 'react';
+
+import {Breadcrumbs, type Crumb} from 'sentry/components/breadcrumbs';
+import ButtonBar from 'sentry/components/buttonBar';
+import FeedbackWidgetButton from 'sentry/components/feedback/widget/feedbackWidgetButton';
+import * as Layout from 'sentry/components/layouts/thirds';
+import {TabList, Tabs} from 'sentry/components/tabs';
+import {t} from 'sentry/locale';
+import {useNavigate} from 'sentry/utils/useNavigate';
+import useOrganization from 'sentry/utils/useOrganization';
+import {
+  type RoutableModuleNames,
+  useModuleURLBuilder,
+} from 'sentry/views/insights/common/utils/useModuleURL';
+import {
+  FRONTEND_LANDING_SUB_PATH,
+  FRONTEND_LANDING_TITLE,
+} from 'sentry/views/insights/pages/frontend/settings';
+import {
+  DOMAIN_VIEW_BASE_URL,
+  OVERVIEW_PAGE_TITLE,
+} from 'sentry/views/insights/pages/settings';
+import {MODULE_TITLES} from 'sentry/views/insights/settings';
+import {ModuleName} from 'sentry/views/insights/types';
+
+type Props = {
+  module?: ModuleName;
+};
+
+// TODO - add props to append to breadcrumbs and change title
+export function FrontendHeader({module}: Props) {
+  const navigate = useNavigate();
+  const {slug} = useOrganization();
+  const moduleURLBuilder = useModuleURLBuilder(true);
+
+  // TODO - we can probably tweek useModuleURLBuilder to also return the base domain view url i.e useDomainViewUrl('frontend', 'http')
+  const frontendBaseUrl = `/organizations/${slug}/${DOMAIN_VIEW_BASE_URL}/${FRONTEND_LANDING_SUB_PATH}/`;
+
+  const crumbs: Crumb[] = [
+    {
+      label: t('Performance'),
+      to: '/performance', // There is no page at `/insights/` so there is nothing to link to
+      preservePageFilters: true,
+    },
+    {
+      label: FRONTEND_LANDING_TITLE,
+      to: frontendBaseUrl,
+      preservePageFilters: true,
+    },
+    {
+      label: module ? MODULE_TITLES[module] : OVERVIEW_PAGE_TITLE,
+      to: undefined,
+      preservePageFilters: true,
+    },
+  ];
+
+  const handleTabChange = (key: ModuleName | typeof OVERVIEW_PAGE_TITLE) => {
+    if (key === module || (key === OVERVIEW_PAGE_TITLE && !module)) {
+      return;
+    }
+    if (!key) {
+      return;
+    }
+    if (key === OVERVIEW_PAGE_TITLE) {
+      navigate(frontendBaseUrl);
+      return;
+    }
+    navigate(moduleURLBuilder(key as RoutableModuleNames));
+  };
+
+  return (
+    <Fragment>
+      <Tabs value={module || OVERVIEW_PAGE_TITLE} onChange={handleTabChange}>
```

Comment:
```suggestion
      <Tabs value={module ?? OVERVIEW_PAGE_TITLE} onChange={handleTabChange}>
```
nit: I like the nullish coalescing operator for these cases to avoid the potential weird case of these expressions returning a boolean

Reply:
Fair! But i'm curious when this would happen, `module` is of type `ModuleName`, wouldn't typescript prevent these weird scenarios?

Reply:
It wouldn't happen in this particular case but I'm averse to using it unless we have to, and have been warned about it in the past. The or operator is perfectly fine here, it just makes me uncomfortable hence the nitpick lol